[hostcfgd] check cached state instead of the next state

yxieca opened the pull request:
**- Why I did it**
'always_enabled' feature can still be disabled/enabled.

**- How I did it**
When checking if a feature is 'always_enabled', check the cached state to prevent new change to be accepted.
Fix an issue where cache value is updated before all the check is done.
Restore 'always_enabled' value in config db if someone wants to change.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

**- How to verify it**
Without the change:
root@str2-dx010-acs-6:/home/admin# show feature status
Feature         State           AutoRestart
--------------  --------------  --------------
...
teamd           always_enabled  enabled
...

root@str2-dx010-acs-6:/home/admin# config feature state  teamd disabled
root@str2-dx010-acs-6:/home/admin# show feature status
Feature         State           AutoRestart
--------------  --------------  --------------
...
teamd           disabled        enabled
...

With the fix:
root@str2-dx010-acs-6:/home/admin# config feature state  teamd disabled
root@str2-dx010-acs-6:/home/admin# show feature status                
Feature         State           AutoRestart
--------------  --------------  --------------
...
teamd           always_enabled  enabled
...

- [ ] 201811
- [x] 201911
- [ ] 202006

abdosi commented:
@yxieca I think we need this once we update sonic-util submodule which will bring change from this PR: https://github.com/Azure/sonic-utilities/pull/1271

yxieca commented:
> @yxieca I think we do not need this change this once we update sonic-util submodule which will bring change from this PR: [Azure/sonic-utilities#1271](https://github.com/Azure/sonic-utilities/pull/1271)
> 
> PR for submodule update: #6070

@abdosi I think we still need this change. I fixed 2 logic issues in this change and added a protection (reverting wrong change). I hope we never need to exercise the protection. But the 2 logic error fixes are necessary.

lguohan reviewed (approved):
extra protection, looks ok.

abdosi reviewed (approved):
LGTM. The change helps "always_enabled" to be maintain and not to be over-written in any case 
as extra check over CLI command.
